aries_vcx/src/utils/mod.rs

```diff
@@ -8,7 +8,6 @@ use crate::{
 };
 
 #[macro_use]
-#[cfg(feature = "vdrtools")]
```

Comment:
This causes aries_vcx to not compile with `cargo build --no-default-features`:

```
error[E0433]: failed to resolve: could not find `indy` in `wallet`
  --> aries_vcx/src/utils/devsetup.rs:12:13
   |
12 |     wallet::indy::{
   |             ^^^^ could not find `indy` in `wallet`

error[E0432]: unresolved imports `aries_vcx_core::wallet::indy`, `aries_vcx_core::WalletHandle`
  --> aries_vcx/src/utils/devsetup.rs:12:13
   |
12 |     wallet::indy::{
   |             ^^^^ could not find `indy` in `wallet`
...
17 |     WalletHandle,
   |     ^^^^^^^^^^^^ no `WalletHandle` in the root

Some errors have detailed explanations: E0432, E0433.
```

I'm not sure when it was removed, but there used to be a CI step for `cargo check`ing many variants of the aries_vcx. It would pick up on stuff like this; are we able to restore that CI step please?

Reply:
Also a general question, i wonder if `devsetup` should be hidden behind `#[test]` too? it seems to just be test helpers

Reply:
I replaced the `cargo check` with some `clippy` jobs, but apparently skipped this one. My bad. Will add a job for this. The funny thing is that running `cargo clippy` on the entire workspace does not catch this - you have to run it on `aries_vcx` alone.

Regarding `devsetup`, this does not belong at all in `aries_vcx` as yeah, it's only used in tests. I'm looking into how to accommodate this right now as moving test utilities requires also moving tests and it's a bigger effort.

Reply:
Okay so putting `devsetup` behind the `vdrtools_wallet` feature seems to solve issues. I plan on taking this soon and decouple test code/utils from production `aries-vcx` as it's only becoming more and more annoying to deal with. Hopefully CI passes now (also added a job for plain `aries-vcx` with no features)

Reply:
In all honesty, probably the way to go would be for `aries-vcx` to have no default features (in which case the job added for this wouldn't be required as this would get caught by the workspace-wide clippy CI job). 

But with the way things are currently I am honestly plain afraid of removing the default features as a lot of tests will start going haywire. So this is perhaps something to consider after the separation between test code and prod code.